web_access_rule_buttons/models.py

```diff
@@ -28,4 +28,11 @@ def check_access_rule_all(self, operations=None):
     return result
 
 
-models.BaseModel.check_access_rule_all = check_access_rule_all
+# Could be any model, we just use a core model to have a 'register_hook'
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    def _register_hook(self, cr):
+        # Add method check_access_rule_all for all models
+        models.BaseModel.check_access_rule_all = check_access_rule_all
```

Comment:
Odoo provides `_patch_method()` for that. See this recent PR that does a similar thing: patches methods in modules based on a condition. Yours should be the same without the condition IMHO. https://github.com/OCA/server-tools/pull/352/files

Thanks for the link, indeed v10 will be an interesting one.